sdk/tables/aztables/tableClient.go

```diff
@@ -7,34 +7,69 @@ import (
 	"context"
 
 	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/runtime"
 )
 
 // A TableClient represents a URL to an Azure Storage blob; the blob may be a block blob, append blob, or page blob.
 type TableClient struct {
-	client *tableClient
-	cred   SharedKeyCredential
+	client  *tableClient
+	service *TableServiceClient
+	cred    SharedKeyCredential
+	name    string
 }
 
 // NewTableClient creates a TableClient object using the specified URL and request policy pipeline.
-func NewTableClient(serviceURL string, cred azcore.Credential, options *TableClientOptions) (*TableClient, error) {
-	con := newConnection(serviceURL, cred, options.getConnectionOptions())
+func NewTableClient(tableName string, serviceURL string, cred azcore.Credential, options *TableClientOptions) (*TableClient, error) {
+	s, err := NewTableServiceClient(serviceURL, cred, options)
+	return s.GetTableClient(tableName), err
+}
+
+func (t *TableClient) Name() string {
+	return t.name
+}
 
-	c, _ := cred.(*SharedKeyCredential)
+// Create creates the table with the name specified in NewTableClient
+func (t *TableClient) Create(ctx context.Context) (*TableResponseResponse, *runtime.ResponseError) {
+	return t.service.Create(ctx, t.name)
+}
+
+// Delete deletes the current table
+func (t *TableClient) Delete(ctx context.Context) (*TableDeleteResponse, *runtime.ResponseError) {
+	return t.service.Delete(ctx, t.name)
+}
 
-	return &TableClient{client: &tableClient{con}, cred: *c}, nil
+// Query queries the tables using the specified QueryOptions
+func (t *TableClient) Query(queryOptions QueryOptions) TableEntityQueryResponsePager {
+	return &tableEntityQueryResponsePager{tableClient: t, queryOptions: &queryOptions, tableQueryOptions: &TableQueryEntitiesOptions{}}
 }
 
-// Create
-func (t TableClient) Create(ctx context.Context, name string) (TableResponseResponse, error) {
-	resp, err := t.client.Create(ctx, TableProperties{&name}, nil, nil)
-	if resp == nil {
-		return TableResponseResponse{}, err
+func (t *TableClient) QueryAsStruct(opt QueryOptions, s FromMapper) StructEntityQueryResponsePager {
```

Comment:
The response type for this operation is mostly just a draft, but I'm curious about any feedback related to this idea for returning query results as a well defined interface struct rather than the default `map[string]interface{}`. This allows developers to use "strongly typed" models rather than the map.

Reply:
The developer would know, typically. Because a table is schema-less it is possible to get back a result with jagged schema. But if that was expected, you'd use the map result query instead.